photutils/detection/findstars.py

```diff
@@ -1,24 +1,63 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
-"""Functions for detecting stars in an astronomical image."""
+"""
+This module implements classes, called Finders, for detecting stars in an
+astronomical image. The general convention is that all Finders are subclasses
+of an abstract class called StarFinder and should be callable classes.
+Additionally, StarFinder defines two abstract methods, namely, find_stars and
+__call__. In general, find_stars implements an algorithm for detecting stars
+and __call__ invokes find_stars to return stars positions estimatives. 
+"""
 
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 from collections import defaultdict
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
+from astropy.utils import deprecated
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+__all__ = ['DAOStarFinder', 'daofind', 'IRAFStarFinder', 'irafstarfind']
 
 
+@deprecated(0.3, alternative='DAOStarFinder')
 def daofind(data, threshold, fwhm, ratio=1.0, theta=0.0, sigma_radius=1.5,
             sharplo=0.2, sharphi=1.0, roundlo=-1.0, roundhi=1.0, sky=0.0,
             exclude_border=False):
+    finder = DAOStarFinder(threshold, fwhm, ratio, theta, sigma_radius,
+                           sharplo, sharphi, roundlo, roundhi, sky,
+                           exclude_border)
+    return finder(data)
+
+
+@deprecated(0.3, alternative='IRAFStarFinder')
+def irafstarfind(data, threshold, fwhm, sigma_radius=1.5, minsep_fwhm=2.5,
+                 sharplo=0.5, sharphi=2.0, roundlo=0.0, roundhi=0.2,
+                 sky=None, exclude_border=False):
+    finder = IRAFStarFinder(threshold, fwhm, sigma_radius, minsep_fwhm,
+                            sharplo, sharphi, roundlo, roundhi, sky,
+                            exclude_border)
+    return finder(data)
+
+
+class StarFinder:
```

Comment:
this should be ``StarFinder(object)``.  You can see more on this at https://wiki.python.org/moin/NewClassVsClassicClass, but the upshot is that for python 2.x code (or 2.x and 3.x-compatible code), you should always use ``class ClassName(object):`` instead of ``class ClassName:``

Also, maybe the class itself should be called ``StarFinderBase`` or something like that?  That makes it clear that it's a base class that should not be used.  Otherwise a new user might be like "hmm, I need to find stars.  Oh, I should use the thing called `StarFinder`!"